taurus/util/impl.py

```diff
@@ -203,24 +219,46 @@ def recursive_flatmap(obj, func, seen=None):
     if isinstance(obj, (list, tuple)):
         for i, x in enumerate(obj):
             if isinstance(x, BaseEntity):
-                res.extend(recursive_flatmap(x, func, seen))
+                res.extend(recursive_flatmap(x, func, seen, chronological))
                 res.extend(func(x))
             else:
-                res.extend(recursive_flatmap(x, func, seen))
+                res.extend(recursive_flatmap(x, func, seen, chronological))
     elif isinstance(obj, dict):
         for x in concatv(obj.keys(), obj.values()):
             if isinstance(x, BaseEntity):
-                res.extend(recursive_flatmap(x, func, seen))
+                res.extend(recursive_flatmap(x, func, seen, chronological))
                 res.extend(func(x))
             else:
-                res.extend(recursive_flatmap(x, func, seen))
+                res.extend(recursive_flatmap(x, func, seen, chronological))
     elif isinstance(obj, DictSerializable):
         for k, x in sorted(obj.__dict__.items()):
-            if isinstance(obj, BaseEntity) and k in obj.skip:
+            if chronological and isinstance(obj, BaseEntity) and k in obj._forward:
+                continue
+            if not chronological and isinstance(obj, BaseEntity) and k in obj.skip:
                 continue
             if isinstance(x, BaseEntity):
-                res.extend(recursive_flatmap(x, func, seen))
+                res.extend(recursive_flatmap(x, func, seen, chronological))
                 res.extend(func(x))
             else:
-                res.extend(recursive_flatmap(x, func, seen))
+                res.extend(recursive_flatmap(x, func, seen, chronological))
     return res
+
+
+def _flattened_sort_order(obj: BaseEntity) -> int:
```

Comment:
This should be consistent with the backend write ordering https://github.com/CitrineInformatics/platform-backend/blob/0b958e2812d6b4e5a7c37ff3d1e2f89776a42c88/code/faraday/model/src/main/scala/io/citrine/faraday/api/model/Storable.scala#L264

Reply:
Is this documented in the API spec, at least as a hint to keep the ordering synchronized between backend and client?

Reply:
No; it is an implicit ordering given other documentation about the way bidirectional links in the data model documentation are bound in the API.